### PR TITLE
Fe notifications page

### PR DESCRIPTION
**Pull request by ChianHuei**
1. moved socket and notification out from Navbar, and added SocketContainer.js
2. a new page displaying past notifications with infiniteScroll
3. integrated socket.io into scrapePriceTag
4. fixed errors in Notification schema
Screen Shot 2020-07-23 at 2 21 46 PM

### CODE REVIEW

File: client/src/components/Notifications.js [hunk 1]
```diff
@@ -45,70 +45,71 @@ const linethrough = {
...
-								<Avatar src={msg.content.image} variant="square" type="image">
-									{cutProductName(msg.content.title)}
-								</Avatar>
-							</ListItemAvatar>
-							<ListItemText>
-								<Typography className={classes.title} color="textPrimary" variant="subtitle2">
-									{cutContentLength(msg.content.title, 30, 'no product title')}
-								</Typography>
-								{msg.content.url ? (
-									<Typography variant="caption" color="textSecondary">
```
Comment (Ethorf) [inline client/src/components/Notifications.js:81]: Looking great! If we could keep these ternaries so my follower notifications don't break, I think we should be able to use the rest the same!

File: client/src/utils/transformText.js [hunk 2]
```diff
@@ -16,4 +16,54 @@ export const convertNumberDecimal=(numberDecimal)=>{
...
+            else if (days < 45) return template("month", 1);
+            else if (days < 365) return template("months", days / 30);
+            else if (years < 1.5) return template("year", 1);
+            else return template("years", years);
+        }
+        return [templates.prefix + compose() + templates.suffix, hours];
+    };
+    
+    return calculateDiff(createdTime)
+}
```
Comment (Ethorf) [inline client/src/utils/transformText.js:69]: Great little function!

**Review by Ethorf — COMMENTED** [review]
Looks great!

File: client/src/components/Notifications.js (hunk 1 above)
Comment (ChianHuei) [reply client/src/components/Notifications.js:81]: Sorry, I didn't notice them. When I was merging, I missed these ternaries. I'm correcting those, so I closed this PR.

**Pull request merged**